### DISCUSSION TITLE
Vision listmode reconstruction

### DISCUSSION THREAD

**Discussion by ChrisMilz** [villekf/OMEGA#43] [Q&A]
Hi, 
I'm currently trying to reconstruct Vision listmode data. However, my only output is a bright (all ones) rectangular area in the lower part of the image. 
I'm using the Vision_main file, all corrections are turned off, implementation 4 with OSEM and improved Siddon's algorithm. 
When running Inveon_PET_main_listmode_example with same parameters (of course with Inveon scanner parameters) the reconstruction works fine.
I assume that reading the listmode file is where things go wrong, because the variable SinM (and the related variables) just contain zeros (which is not the case when reconstructing the Inveon PET data)

Maybe you have some ideas where the issue might lie and what to do about. I'd appreciate any suggestions.

thanks in advance
Chris

**Comment by villekf** [comment]
Hi,

I'm also guessing that the issue is with the reading of the listmode data. Does it take a while to load the data or is it practically instantaneous? I'm a little surprised that the data is zeros and not garbage, which suggests that the listmode data is not actually loaded.

**Reply by ChrisMilz** [reply]
It's not instantaneous, but takes ~15s according to the output printed by load_data_Vision.m.

**Reply by villekf** [reply]
Clearly it seems to read the file correctly. The listmode works with Siemens software? Is it a .ptd-file? There is a certain bit that controls whether the 64-bit event is read so it is possible that either that bit is always set such that the event is not read, or there is possibly an offset of one or more bits. Without the data itself there isn't much I can do, nor can I release the source codes. If you can share the listmode data I can try to look it up at some point.
